packages/fxa-payments-server/src/routes/Subscriptions/index.test.tsx

```diff
@@ -1,106 +1,596 @@
+
+import { PAYMENT_ERROR_1 } from '../../lib/errors';
 import {
-  AppContext,
-  AppContextType,
-  defaultAppContext,
-} from '../../lib/AppContext';
-import { defaultConfig } from '../../lib/config';
-import { Customer, Profile, Subscription } from '../../store/types';
-
-beforeEach(() => {});
-
-afterEach(cleanup);
-
-const mockProfile: Profile = {
-  amrValues: [],
-  avatar: 'avatar',
-  avatarDefault: true,
-  displayName: null,
-  email: 'email',
-  locale: 'locale',
-  twoFactorAuthentication: false,
-  uid: 'uid',
-};
-const mockCustomer: Customer = {
-  payment_type: 'cc',
-  last4: '4444',
-  exp_month: '12',
-  exp_year: '22',
-  subscriptions: [],
-};
-const mockSubscription: Subscription = {
-  subscriptionId: 'abc',
-  cancelledAt: null,
-  createdAt: Date.now(),
-  productName: 'pro jest',
-};
-const mockPlan = {
-  plan_id: 'abc',
-  plan_name: 'abc',
-  product_id: 'abc',
-  product_name: 'abc',
-  currency: 'abc',
-  amount: 100,
-  interval: 'abc',
-};
-const mockCustomerSubscription = {
-  cancel_at_period_end: false,
-  current_period_end: 99,
-  current_period_start: 9,
-  end_at: null,
-  nickname: 'abc',
-  plan_id: 'abc',
-  status: 'abc',
-  subscription_id: 'abc',
-};
-const mockedSubscriptionsProps = {
-  profile: { error: null, loading: false, result: mockProfile },
-  plans: { error: null, loading: false, result: [mockPlan] },
-  customer: { error: null, loading: false, result: mockCustomer },
-  subscriptions: { error: null, loading: false, result: [mockSubscription] },
-  customerSubscriptions: [mockCustomerSubscription],
-  fetchSubscriptionsRouteResources: jest.fn(),
-  cancelSubscription: jest.fn(),
-  cancelSubscriptionStatus: {
-    error: null,
-    loading: false,
-    result: mockSubscription,
-  },
-  resetCancelSubscription: jest.fn(),
-  reactivateSubscription: jest.fn(),
-  reactivateSubscriptionStatus: { error: null, loading: false, result: null },
-  resetReactivateSubscription: jest.fn(),
-  updatePayment: jest.fn(),
-  updatePaymentStatus: { error: null, loading: false, result: null },
-  resetUpdatePayment: jest.fn(),
-};
-const Subject = (props: SubscriptionsProps) => {
-  const config = defaultConfig();
-  const appContextValue = {
-    ...defaultAppContext,
-    config: {
-      ...config,
-      servers: {
-        ...config.servers,
-        content: {
-          url: 'http://127.0.0.1:3030'
-        }
-      }
-    },
-    queryParams: { successfulSupportTicketSubmission: 'quux' },
+  wait,
+  defaultAppContextValue,
+  MockApp,
+  setupMockConfig,
+  mockConfig,
+  mockServerUrl,
+  mockOptionsResponses,
+  mockStripeElementOnChangeFns,
+  elementChangeResponse,
+  STRIPE_FIELDS,
+  VALID_CREATE_TOKEN_RESPONSE,
+  MOCK_PROFILE,
+  MOCK_PLANS,
+  MOCK_ACTIVE_SUBSCRIPTIONS,
+  MOCK_CUSTOMER,
+} from '../../lib/test-utils';
+
+import { SettingsLayout } from '../../components/AppLayout';
+import Subscriptions from './index';
+
+describe('routes/Subscriptions', () => {
```

Comment:
This is the main event: Tests for the subscription management route with mocked server APIs.